### PR TITLE
Bigint

### PR DESCRIPTION
**Pull request by briandowns**
### Well detailed description of the change :

Add support for bigint data type, work done by @jamsilva .

Resolves: #

#

### Type of change:

- [ ] Bug fix

- [x] New feature

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#

### Housekeeping

- [ ] Tests have been updated to reflect the changes done within this PR (if applicable).

- [ ] Documentation has been updated to reflect the changes done within this PR (if applicable).

#

### CODE REVIEW

**Comment by briandowns** [comment]
The failures seem to be false positives. They run on my Ubuntu and Mac without issue.

**Comment by Jason2605** [comment]
Interesting yeah, seems to be the HTTP tests timing out - will need to investigate

**Comment by Jason2605** [comment]
This looks awesome - sorry been away for a bit and just trying to get back into the swing of things. Will probably need to look at getting some docs and tests in for this as well

**Comment by briandowns** [comment]
Looks like httpbin.org is running slow from github.

**Comment by briandowns** [comment]
These still seem to be http failures.

**Comment by Jason2605** [comment]
Sorry been slow on this one, will try get it sorted, if not we can just ignore the failing HTTP tests for this merge

**Comment by briandowns** [comment]
Looks like we're good on CI checks again!

File: src/optionals/bigint.c [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+    } else if (isBigIntValue(result->value)) {
+        return result->value;
+    } else {
+        return EMPTY_VAL;
+    }
+}
+
+static Value newBigInt(DictuVM *vm, int argCount, Value *args) {
+    if (argCount > 1) {
+        return newResultError(vm, "error: wrong number of arguments");
```
Comment (Jason2605) [inline src/optionals/bigint.c:181]: I would have this be a runtime error

**Comment by briandowns** [comment]
More HTTP failures. I'd restart but I don't have that option to do so in the interface.

**Comment by Jason2605** [comment]
Thanks for this

**Pull request merged**